src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -27,34 +28,31 @@ uint64_t HSHomeObject::get_sequence_num_from_shard_id(uint64_t shard_id_t) {
     return shard_id_t & (max_shard_num_in_pg() - 1);
 }
 
-std::string HSHomeObject::serialize_shard(const Shard& shard) const {
+std::string HSHomeObject::serialize_shard_info(const ShardInfo& info) {
     nlohmann::json j;
-    j["shard_info"]["shard_id"] = shard.info.id;
-    j["shard_info"]["pg_id_t"] = shard.info.placement_group;
-    j["shard_info"]["state"] = shard.info.state;
-    j["shard_info"]["created_time"] = shard.info.created_time;
-    j["shard_info"]["modified_time"] = shard.info.last_modified_time;
-    j["shard_info"]["total_capacity"] = shard.info.total_capacity_bytes;
-    j["shard_info"]["available_capacity"] = shard.info.available_capacity_bytes;
-    j["shard_info"]["deleted_capacity"] = shard.info.deleted_capacity_bytes;
-    j["ext_info"]["chunk_id"] = shard.chunk_id;
+    j["shard_info"]["shard_id_t"] = info.id;
+    j["shard_info"]["pg_id_t"] = info.placement_group;
+    j["shard_info"]["state"] = info.state;
+    j["shard_info"]["created_time"] = info.created_time;
+    j["shard_info"]["modified_time"] = info.last_modified_time;
+    j["shard_info"]["total_capacity"] = info.total_capacity_bytes;
+    j["shard_info"]["available_capacity"] = info.available_capacity_bytes;
+    j["shard_info"]["deleted_capacity"] = info.deleted_capacity_bytes;
     return j.dump();
 }
 
-Shard HSHomeObject::deserialize_shard(const std::string& shard_json_str) const {
-    auto shard_json = nlohmann::json::parse(shard_json_str);
+ShardInfo HSHomeObject::deserialize_shard_info(const char* json_str, size_t str_size) {
     ShardInfo shard_info;
-    shard_info.id = shard_json["shard_info"]["shard_id"].get< shard_id_t >();
+    auto shard_json = nlohmann::json::parse(json_str, json_str + str_size);
+    shard_info.id = shard_json["shard_info"]["shard_id_t"].get< shard_id_t >();
     shard_info.placement_group = shard_json["shard_info"]["pg_id_t"].get< pg_id_t >();
     shard_info.state = static_cast< ShardInfo::State >(shard_json["shard_info"]["state"].get< int >());
     shard_info.created_time = shard_json["shard_info"]["created_time"].get< uint64_t >();
     shard_info.last_modified_time = shard_json["shard_info"]["modified_time"].get< uint64_t >();
     shard_info.available_capacity_bytes = shard_json["shard_info"]["available_capacity"].get< uint64_t >();
     shard_info.total_capacity_bytes = shard_json["shard_info"]["total_capacity"].get< uint64_t >();
     shard_info.deleted_capacity_bytes = shard_json["shard_info"]["deleted_capacity"].get< uint64_t >();
-    auto shard = Shard(shard_info);
-    shard.chunk_id = shard_json["ext_info"]["chunk_id"].get< uint16_t >();
-    return shard;
+    return shard_info;
 }
 
 ShardManager::Result< ShardInfo > HSHomeObject::_create_shard(pg_id_t pg_owner, uint64_t size_bytes) {
```

Comment:
This has to AsyncResult. Blob manager changed it in another PR.

Reply:
updated.